pkg/object/function/provider/provider.go

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sresource "k8s.io/apimachinery/pkg/api/resource"
+	k8sapisv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/client/pkg/kn/commands"
+	clientservingv1 "knative.dev/client/pkg/serving/v1"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/object/function/spec"
+	"github.com/megaease/easegress/pkg/supervisor"
+)
+
+type (
+	// FaaSProvider is the physical serverless function instance manager
+	FaaSProvider interface {
+		Init() error
+		GetStatus(name string) (*spec.Status, error)
+		Create(funcSpec *spec.Spec) error
+		Delete(name string) error
+		Update(funcSpec *spec.Spec) error
+	}
+
+	// knativeClient is client for communicating with Knative type FaaSProvider
+	knativeClient struct {
+		superSpec     *supervisor.Spec
+		serviceClient clientservingv1.KnServingClient
+		namespace     string
+		timeOut       time.Duration
```

Comment:
```suggestion
		timeout       time.Duration
```

to align with other `timeout`s.